tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/ControllerActionInvoker_InvokeAction_Integration.cs

```diff
@@ -94,17 +95,28 @@ public class ControllerActionInvoker_InvokeAction_Integration
                     {
                         var context = HttpContext.Current;
                         var scope = SharedItems.TryPeekScope(context, AspNetMvcIntegration.HttpContextKey);
-                        var securityTransport = new SecurityCoordinator(security, context, scope.Span);
-                        if (!securityTransport.IsBlocked)
+                        if (scope is not null)
                         {
-                            var inputData = new Dictionary<string, object> { { AddressesConstants.ResponseBody, ObjectExtractor.Extract(responseObject) } };
-                            securityTransport.CheckAndBlock(inputData);
+                            var securityTransport = new SecurityCoordinator(security, context, scope.Span);
+                            if (!securityTransport.IsBlocked)
+                            {
+                                var extractedObject = ObjectExtractor.Extract(responseObject);
+                                if (extractedObject is not null)
+                                {
+                                    var inputData = new Dictionary<string, object> { { AddressesConstants.ResponseBody, extractedObject } };
+                                    securityTransport.CheckAndBlock(inputData);
+                                }
+                            }
+                        }
+                        else
+                        {
+                            Log.Warning("Scope was null in ControllerActionInvoker_InvokeAction_Integration.OnMethodEnd, cannot check security");
```

Comment:
I wonder if we should add this or not 🤔 I guess we don't _ever_ expect this to happen, right? My concern is that we don't _know_ if it will happen or not in some cases, so we could end up flooding the logs? If so, should it be Debug?

Reply:
I changed them to debug, I guess we would like to know that there's no scope but this could be a telemetry one later rmaybe